docs/pages/debugging/create-devtools-plugins.mdx

```diff
@@ -0,0 +1,115 @@
+
+You create a dev tools plugin, whether that's for inspecting aspects of a common framework or library or something specific to your custom code.
+
+A plugin consists of three key elements:
+- an Expo app to display the dev tools web user interface,
+- an **expo-module.config.json** for Expo CLI recognition,
+- calls to `expo/devtools` API for the app to communicate back and forth with the dev tool's web interface.
+
+Plugins can be included inside your app's monorepo or distributed separately on npm.
+
+## Create a plugin
+
+### Create an Expo app
+
+The web interface of your dev tool will be an Expo app. Create a new blank Expo app:
+
+<Terminal cmd={['$ npx create-expo-app']} />
+
+Then, add the packages required to build the app for the web and to use the dev tools API:
+
+<Terminal cmd={['$ npx expo install react-native-web react-dom']} />
+
+### Update package.json
+
+Move any dependencies from `dependencies` to `devDependences` in **package.json**. These dependencies are needed for building the web version of the dev tools UI, and are part of your app's development. Having them in `devDependencies` ensures they are not included in your app.
+
+```diff package.json
+"dependencies": {
+-  "expo": "~50.0.0",
+-  "react": "18.2.0",
+-  "react-dom": "18.2.0",
+-  "react-native": "0.73.0",
+-  "react-native-web": "~0.19.6"
+},
+"devDependencies": {
+  "@babel/core": "^7.20.0",
+  "@expo/webpack-config": "^18.1.2",
++  "expo": "~50.0.0",
++  "react": "18.2.0",
++  "react-dom": "18.2.0",
++  "react-native": "0.73.0",
++  "react-native-web": "~0.19.6"
+},
+```
+
+Add a `homepage` field to define the exported base URL for web in the format of `/_expo/plugins/{pluginName}`. For example:
+
+```diff package.json
+  "name": "expo-plugin-helloworld",
+  "version": "1.0.0",
++  "homepage": "/_expo/plugins/expo-plugin-helloworld",
+```
+
+### Add message sending and receiving code to the plugin's web UI
+
+`getDevToolsPluginClientAsync` initiates a client that can send messages and listen for incoming messages:
+
+```js
+  import { getDevToolsPluginClientAsync } from 'expo/devtools';
+
+  const client = await getDevToolsPluginClientAsync('REPLACE-FOR-YOUR-PLUGIN-NAME');
+
+  client.sendMessage('ping');
+  client.addMessageListener('pong', (data) => {
+    console.log('receiving pong', data);
+  });
+```
+
+A `useDevToolsPluginClient` hook is also available to provide a client entirely within a React component.
+
+Client apps using your plugin will also be able to communicate using `getDevToolsPluginClientAsync` or `useDevToolsPluginClient`. The identifier passed to `getDevToolsPluginClientAsync` must match in both the plugin and the client app.
+
+### Build the app for web
+
+Create a **package.json** script that builds the web app and moves it to the **dist** directory. This will be served by the Expo CLI when invoking the plugin.
```

Comment:
```suggestion
Create a **package.json** script to build the web app and move it to the **dist** directory. This will be served by the Expo CLI when invoking the plugin.
```